### PR TITLE
Improve BOLD schemas

### PR DESCRIPTION
**Pull request by cris-santos**
## Summary

- Replace no-op `z.string()` overrides in Bold `.extend()` calls with `NonEmptyStringSchema` for event names, attribute names, document categories, subtypes, and relation types
- Trim test suites to meaningful domain-specific assertions only (enum validation, BigNumber edge cases, nonempty arrays, looseObject passthrough)
- Remove tests that only verify default Zod behavior (required field rejection, basic valid input acceptance)

## Test plan

- [x] TypeScript compiles cleanly for both `shared-types` and `shared-methodologies-bold-types`
- [x] Lint passes for both projects
- [x] All 1860 tests pass across both suites

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Summary by CodeRabbit

* **New Features**
  * Added runtime validation for large numeric strings (including scientific notation) and stricter enum/value schemas.
  * Schemas now accept and preserve extra unknown fields (loose/passthrough behavior).

* **Bug Fixes**
  * Required document and event name/category/type fields now must be non-empty.
  * Enforced non-empty arrays and tighter percentage/number validations.

* **Tests**
  * Expanded test coverage for schemas, enums, string formats, rewards distribution, and validation edge cases.

### CODE REVIEW

**Comment by cris-santos** [comment]
@coderabbitai review

**Review by sangalli — APPROVED** [review]

File: libs/shared/types/src/methodology/methodology-participant.types.ts [hunk 1]
```diff
@@ -18,7 +18,6 @@ export const MethodologyParticipantSchema = z.looseObject({
   piiSnapshotId: NonEmptyStringSchema,
   taxId: NonEmptyStringSchema,
   taxIdType: NonEmptyStringSchema,
-  // TODO: replace with MethodologyParticipantTypeSchema once migrated to Zod
```
Comment (sangalli) [inline libs/shared/types/src/methodology/methodology-participant.types.ts:21]: Should we actually implement a schema MethodologyParticipantTypeSchema?
